core-plugins/src/main/java/io/cdap/plugin/batch/source/ExcelInputFormat.java

```diff
@@ -175,6 +176,8 @@ public void initialize(InputSplit genericSplit, TaskAttemptContext context) thro
             isStreaming = true;
             break;
           case OLE2:
+            // workaround for large files
+            IOUtils.setByteArrayMaxOverride(Integer.MAX_VALUE / 2);
```

Comment:
Check if setting it to a higher value is not creating any heap memory issues. Test it on a cluster with nodes having smaller RAM(e.g. 2GB).

Reply:
@psainics what is the max size of xls file we have tested with this

Reply:
330 MB